firefox-ios/firefox-ios-tests/Tests/XCUITests/DesktopModeTests.swift

```diff
@@ -217,5 +231,44 @@ class DesktopModeTestsIphone: IphoneOnlyTestCase {
         waitUntilPageLoad()
         XCTAssert(app.webViews.staticTexts.matching(identifier: "DESKTOP_UA").count > 0)
     }
+
+    func testLongPressReload_tabTrayExperimentOn() {
+        addLaunchArgument(jsonFileName: "defaultEnabledOn", featureName: "tab-tray-ui-experiments")
+        app.launch()
+        if skipPlatform { return }
+        navigator.openURL(path(forTestPage: "test-user-agent.html"))
+        waitUntilPageLoad()
+        mozWaitForElementToExist(app.webViews.staticTexts.firstMatch)
+        XCTAssert(app.webViews.staticTexts.matching(identifier: "MOBILE_UA").count > 0)
+
+        navigator.nowAt(BrowserTab)
+        if #unavailable(iOS 16) {
+            // iOS 15 displays a toast that covers the reload button
+            sleep(2)
+        }
+        navigator.goto(ReloadLongPressMenu)
+        navigator.performAction(Action.ToggleRequestDesktopSite)
+        waitUntilPageLoad()
+        XCTAssert(app.webViews.staticTexts.matching(identifier: "DESKTOP_UA").count > 0)
+
+        // Covering scenario that when reloading the page should preserve Desktop site
+        navigator.nowAt(BrowserTab)
+        navigator.performAction(Action.ReloadURL)
+        waitUntilPageLoad()
+        XCTAssert(app.webViews.staticTexts.matching(identifier: "DESKTOP_UA").count > 0)
+
+        navigator.performAction(Action.OpenNewTabFromTabTray)
+        // The experiment is not opening the keyboard on a new tab
```

Comment:
Note that I added this comment `// The experiment is not opening the keyboard on a new tab` where we were checking the keyboard was opened when opening a new tab. With the tab tray UI experiment we don't open the keyboard (focusing on the URL bar) when we open a new tab.